go-controller/pkg/node/udn_isolation.go

```diff
@@ -368,73 +368,106 @@ func (m *UDNHostIsolationManager) updateUDNPodIPs(namespacedName string, podIPs
 	}
 
 	// update internal state only after successful transaction
-	m.udnPodIPsv4.updatePodIPsAfterTX(namespacedName, newV4IPs)
-	m.udnPodIPsv6.updatePodIPsAfterTX(namespacedName, newV6IPs)
+	m.udnPodIPsv4.updatePodElementsAfterTX(namespacedName, newV4IPs)
+	m.udnPodIPsv6.updatePodElementsAfterTX(namespacedName, newV6IPs)
 	return nil
 }
 
-// nftPodIPSet is a helper struct to manage pod IPs in an nftables set.
-// Can be used for ipv4 and ipv6 sets.
-type nftPodIPSet struct {
+// nftPodElementsSet is a helper struct to manage an nftables set with pod-owned elements.
+// Can be used to store pod IPs, or more complex elements.
+type nftPodElementsSet struct {
 	setName string
-	// podName: ips
-	podsIPs map[string]sets.Set[string]
+	// podName: set elements
+	podElements map[string]sets.Set[string]
+	// podIPs may be reused as soon as the pod reaches Terminating state, and delete event may come later.
```

Comment:
I think this statement is a little backwards. The delete event will come immediately, and OVNK will release the IP without waiting for termination grace period (this is probably a bug). Since the pod informer is serialized you should get the delete pod, before another new pod event can take its IP. I'm not sure what your exact scenario is, but to protect against races with this stuff we use the logical port cache in the default network controller and have an expiry timer to indicate when a pod has been deleted.

Reply:
oh you are right, my bad